### PR TITLE
DBZ-2278 Resolve concurrent access to the thread map

### PR DESCRIPTION
**Pull request by jpechane**
https://issues.redhat.com/browse/DBZ-2278

### CODE REVIEW

File: debezium-connector-mysql/src/main/java/io/debezium/connector/mysql/BinlogReader.java [hunk 1]
```diff
@@ -116,7 +115,7 @@
     private Heartbeat heartbeat;
     private MySqlJdbcContext connectionContext;
     private final float heartbeatIntervalFactor = 0.8f;
-    private final Set<Thread> binaryLogClientThreads = Collections.synchronizedSet(new HashSet<>(2));
+    private final Map<String, Thread> binaryLogClientThreads = new ConcurrentHashMap<>(2);
```
Comment (gunnarmorling) [inline debezium-connector-mysql/src/main/java/io/debezium/connector/mysql/BinlogReader.java:118]: You could keep it as a set, using `newSetFromMap()`:

```suggestion
    private final Map<String, Thread> binaryLogClientThreads = Collections.newSetFromMap(new ConcurrentHashMap<>(3));
```

Also changed the initial size, as 2 wouldn't be enough to hold two elements actually without a resize when using the default load factor. Not that it makes any difference (I'd just call the parameterless constructor), but if we give a size explicitly, it should be large enough.

Comment (jpechane) [reply debezium-connector-mysql/src/main/java/io/debezium/connector/mysql/BinlogReader.java]: @gunnarmorling Modified

**Comment by gunnarmorling** [comment]
This needs formatting. Thx!

**Comment by jpechane** [comment]
@gunnarmorling Done, the code formatter is becoming sadist ;-)

**Comment by gunnarmorling** [comment]
I'm wondering whether we should not take it to the next level and have CI apply the formatter and automatically commit any changes, if needed. WDYT, @jpechane, @Naros?

**Comment by jpechane** [comment]
@gunnarmorling I am not fully convinced on this. It might make people confused when they are asked to update their PRs etc.
What I'd rather see is a separate github actions task that would check the formatting so it is clearly visible at the start that there is a formatting issue.

**Comment by gunnarmorling** [comment]
Ok, good idea. Captured here: https://issues.redhat.com/browse/DBZ-2287.

>

**Pull request merged**